src/v/security/audit/audit_log_manager.h

```diff
@@ -205,6 +205,10 @@ class audit_log_manager
     /// audit topic.
     bool _auth_misconfigured{false};
 
+    /// Represents whether the feature is actually active, not the
+    /// representation of the config variable
+    bool _effectively_enabled{false};
+
```

Comment:
does this factor into whether or not audit messages are enqueued or not?

Reply:
As it stands, no. I'm still on the fence about if it should or not

Reply:
I _feel_ like it should... what's the argument against it?  If it's "effectively" disabled, I'd say we don't enqueue and return true?

Reply:
well it could be effectively disabled but the audit switch is ON, and maybe the client thought "oh its enabled i assume from this point on messages are now being audited". I guess we could just assert in the docs its enabled when u see a relevant log line and it may take some time.

Maybe it is something we should address as proposed. The common case is the node starting up and then enabling auditing and the kafka client taking some time to complete the initialization phase, which sets up credentials and attempts to make the audit topic. If it cannot create the audit topic and its not already up, it will forever retry this, which means the switch will be in the ON state but effectively enabled is false for a long period of time.